src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1029,7 +1061,11 @@ trait Contexts { self: Analyzer =>
      *  if any such symbol is accessible from this context.
      */
     private def importedAccessibleSymbol(imp: ImportInfo, name: Name, requireExplicit: Boolean, record: Boolean): Symbol =
-      imp.importedSymbol(name, requireExplicit, record) filter (s => isAccessible(s, imp.qual.tpe, superAccess = false))
+      if (isExcludedRootImport(imp)) NoSymbol
+      else imp.importedSymbol(name, requireExplicit, record) filter (s => isAccessible(s, imp.qual.tpe, superAccess = false))
+
+    private def isExcludedRootImport(imp: ImportInfo): Boolean =
+      imp.isRootImport && excludedRootImportsCached.get(unit).map(_.contains(imp.qual.symbol)).getOrElse(false)
```

Comment:
`.map(f).getOrElse(false)` could be `.exists(f)`

Reply:
I had that during one version while trying to get everything working... :)